### PR TITLE
Copter: add THROW_NEXTMODE parameter and switch to mode after throw completes

### PR DESCRIPTION
**Pull request by rmackay9**
This PR adds a new parameter THROW_NEXTMODE and some related functions to allow the vehicle to switch into a specified mode (like AUTO) after the throw is completed.

I actually haven't flight tested this yet but will do so probably tomorrow.

### CODE REVIEW

File: ArduCopter/Parameters.cpp [hunk 1]
```diff
@@ -973,7 +973,14 @@ const AP_Param::GroupInfo ParametersG2::var_info[] = {
     // @Group: BTN_
     // @Path: ../libraries/AP_Button/AP_Button.cpp
     AP_SUBGROUPINFO(button, "BTN_", 2, ParametersG2, AP_Button),
-    
+
+    // @Param: THROW_NEXTMODE
+    // @DisplayName: Throw mode's follow up mode
+    // @Description: Vehicle will switch to this mode after the throw is successfully completed.  Default is to stay in throw mode (18)
+    // @Values: 3:Auto,4:Guided,6:RTL,9:Land,17:Brake
```
Comment (OXINARF) [inline ArduCopter/Parameters.cpp]: We should add Throw here too since that's the default and a valid value

File: ArduCopter/Copter.h [hunk 2]
```diff
@@ -274,6 +274,11 @@ class Copter : public AP_HAL::HAL::Callbacks {
         uint32_t start_ms;
     } takeoff_state;
 
+    // throw mode state
+    struct {
+        ThrowModeState state;
```
Comment (OXINARF) [inline ArduCopter/Copter.h]: How about naming this _mode_? `throw_state.state` is strange.

Comment (rmackay9) [inline ArduCopter/Copter.h]: how about we rename "ThrowModeState state" to "ThrowModeStage stage"?

File: ArduCopter/Parameters.cpp (hunk 1 above)
Comment (rmackay9) [inline ArduCopter/Parameters.cpp]: @OXINARF, yes, good point, thanks!

File: ArduCopter/Copter.h (hunk 2 above)
Comment (OXINARF) [inline ArduCopter/Copter.h]: That looks good

**Comment by rmackay9** [comment]
This incorporates Chambana's changes from this PR.  https://github.com/ArduPilot/ardupilot/pull/4410

**Comment by rmackay9** [comment]
Closing for now as I'm making a number of updates after flight testing (which uncovered some issues)

**Pull request closed without merging**